### ISSUE TITLE
Hail 0.2 cloudtools clusters fail to connect to Jupyter kernel

### ISSUE THREAD

**Issue by danking** [hail-is/hail#5505]
Appears unrelated to hail version. JAR and ZIP:
```
gs://hail-common/builds/0.2/jars/hail-0.2-3b1cb0772301-Spark-2.2.0.jar
gs://hail-common/builds/0.2/python/hail-0.2-3b1cb0772301.zip
```

In Google Chrome we see 404s for
```
GET http://localhost:8123/spark/api/v1/applications
```
which happened repeatedly if you try to evaluate a cell.

On the leader node of the spark cluster, `journalctl -u jupyter` shows:
```
-- Logs begin at Fri 2019-03-01 19:54:49 UTC, end at Fri 2019-03-01 20:11:51 UTC. --
Mar 01 19:59:03 dk-m systemd[1]: Started Jupyter Notebook.
Mar 01 19:59:04 dk-m python[5149]: [I 19:59:04.630 NotebookApp] Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
Mar 01 19:59:04 dk-m python[5149]: [W 19:59:04.796 NotebookApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
Mar 01 19:59:04 dk-m python[5149]: [D 19:59:04.802 NotebookApp] Paths used for configuration of jupyter_notebook_config:
Mar 01 19:59:04 dk-m python[5149]:             /etc/jupyter/jupyter_notebook_config.json
Mar 01 19:59:04 dk-m python[5149]: [D 19:59:04.803 NotebookApp] Paths used for configuration of jupyter_notebook_config:
Mar 01 19:59:04 dk-m python[5149]:             /usr/local/etc/jupyter/jupyter_notebook_config.json
Mar 01 19:59:04 dk-m python[5149]: [D 19:59:04.804 NotebookApp] Paths used for configuration of jupyter_notebook_config:
Mar 01 19:59:04 dk-m python[5149]:             /opt/conda/etc/jupyter/jupyter_notebook_config.json
Mar 01 19:59:04 dk-m python[5149]: [D 19:59:04.804 NotebookApp] Paths used for configuration of jupyter_notebook_config:
Mar 01 19:59:04 dk-m python[5149]:             /root/.jupyter/jupyter_notebook_config.json
Mar 01 19:59:04 dk-m python[5149]: [W 19:59:04.904 NotebookApp] Error loading server extension jupyter_spark
Mar 01 19:59:04 dk-m python[5149]:     Traceback (most recent call last):
Mar 01 19:59:04 dk-m python[5149]:       File "/opt/conda/lib/python3.6/site-packages/notebook/notebookapp.py", line 1575, in init_server_extensions
Mar 01 19:59:04 dk-m python[5149]:         func(self)
Mar 01 19:59:04 dk-m python[5149]:       File "/opt/conda/lib/python3.6/site-packages/jupyter_spark/__init__.py", line 30, in load_jupyter_server_extension
Mar 01 19:59:04 dk-m python[5149]:         from .handlers import SparkHandler
Mar 01 19:59:04 dk-m python[5149]:       File "/opt/conda/lib/python3.6/site-packages/jupyter_spark/handlers.py", line 8, in <module>
Mar 01 19:59:04 dk-m python[5149]:         class SparkHandler(IPythonHandler):
Mar 01 19:59:04 dk-m python[5149]:       File "/opt/conda/lib/python3.6/site-packages/jupyter_spark/handlers.py", line 13, in SparkHandler
Mar 01 19:59:04 dk-m python[5149]:         @tornado.web.asynchronous
Mar 01 19:59:04 dk-m python[5149]:     AttributeError: module 'tornado.web' has no attribute 'asynchronous'
```

It appears that Jupyter starts even though one of its plugins fail. This sucks, since Jupyter doesn't actually seem to work if jupyter_spark fails. This happens even if the evaluated cell contains only `3 + 3`.

**Comment by danking** [comment]
Seems to have been a known issue that was not resolved. https://github.com/jupyter/notebook/issues/4311

**Comment by danking** [comment]
Tornado 6.0 was released 5 hours ago. https://pypi.org/project/tornado/#history

**Comment by akotlar** [comment]
amazing timing

**Comment by danking** [comment]
Appears that Jupyter [fixed this](https://github.com/jupyter/notebook/pull/4310/files) but [did not publish to PyPI](https://pypi.org/project/notebook/#history) after the fix. I guess we can insert an explicit tornado max version number into init_notebook1.py until jupyter fixes themselves.

**Comment by apampana** [comment]
```
```I think its not identifying the scripts too from the folder. Operation completed over 1 objects/538.0 B.                                      
Can't locate CGI.pm in @INC (you may need to install the CGI module) (@INC contains: /vep/loftee /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at /vep/variant_effect_predictor/variant_effect_predictor.pl line 46.
BEGIN failed--compilation aborted at /vep/variant_effect_predictor/variant_effect_predictor.pl line 46. ```

**Comment by danking** [comment]
Fix has been deployed and I confirmed it resolves the issue in a freshly created cluster.

**Comment by danking** [comment]
Fixed now with nbconvert >=5.4.1 and notebook >=5.7.5.

https://github.com/jupyter/notebook/issues/4311#issuecomment-470110129
